pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/jni/IValueUtils.java

```diff
@@ -207,9 +235,66 @@ public static NDList forward(PtSymbolBlock block, NDList inputs, boolean isTrain
                 inputs.stream()
                         .map(input -> ((PtNDArray) input).getHandle())
                         .toArray(Pointer[]::new);
-
-        Pointer result = PyTorchLibrary.LIB.moduleForward(block.getHandle(), arrayHandles, isTrain);
+        String[] names = inputs.stream().map(NDArray::getName).toArray(String[]::new);
+        Pointer[] iValueInputs = getInputs(arrayHandles, names);
+        Pointer result = PyTorchLibrary.LIB.moduleForward(block.getHandle(), iValueInputs, isTrain);
         PtNDManager manager = (PtNDManager) inputs.get(0).getManager();
         return forwardHelper(result, manager);
     }
+
+    private static boolean isNameList(String name) {
+        return Pattern.matches("\\w+\\[]", name);
+    }
+
+    private static boolean isNameDict(String name) {
+        return name.contains(".");
+    }
+
+    private static Pointer[] getInputs(Pointer[] arrays, String[] names) {
+        List<PairList<String, Pointer>> outputs = new ArrayList<>();
+        Map<String, Integer> indexMap = new ConcurrentHashMap<>();
+        for (int i = 0; i < arrays.length; i++) {
```

Comment:
In our assumption, is this array has to be a Map<>, a List or single inputs? If this is the case, the `arrays` can just match one pattern and we do all the conversion there.